### ISSUE TITLE
Selecting the "other" option in plant community action does not prompt the user to enter a name -- displaying "other" is not complete

### ISSUE THREAD

**Issue by LisaMoore1** [bcgov/range-web#169]
**REQUIREMENTS**
- [ ] user can easily enter text “other” text into field and intuitively save/move to next field

**Comment by LisaMoore1** [comment]
@calebissharp This is great figuring out how handing issues between us works.  Statuses that matter to me:
- [x] when someone has looked at the issue (it has person and size)
- [x] when person working on issue has a resolution (this is where "ready to test" is being used now)
- [ ]  when something is in TEST environment so I can test 
- [x] when something I tested needs to be looked at again (label: unresolved)

Ideas to deal with third bullet above? Would it work to save "ready to test" until it is actually available in TEST for me? Do we need something that says something is resolved but not in TEST yet? Knowing myself, if I see easily what I can test RIGHT NOW I'm more likely to get it done right now. If I'm not sure if it is really in test or not I'm more likely to just wait until later...perhaps indefinitely. :)

**Comment by calebissharp** [comment]
I would be fine with waiting to add the label until the changes are actually deployed to test. I think the biggest issue is I also don't know when something is finished deploying either. @micheal-w-wells may be able to set up a bot in our slack channel that will notify when a deployment is finished, and that will let me know to go add the labels.

**Comment by LisaMoore1** [comment]
Looking at @micheal-w-wells note re:
1.  mark someone as a reviewer for a PR
2.  once it is reviewed you merge it in yourself
3.  code auto-rolls to test once merged (with 25 minute wait)
Would it work if whoever does the merge labels the related issues "ready to test" immediately after the merge I will know I should be ready to test 25 minutes after time of label add.

**Comment by calebissharp** [comment]
As long as you're ok with keeping track of the time, this way is fine with me. If not, we should get notifications working soon and I can add the labels when the changes are actually deployed.

**Comment by LisaMoore1** [comment]
I’m good with just looking at the time.  Let’s take that approach for now.

**Comment by LisaMoore1** [comment]
@calebissharp Like the indicator plant one...

The user needs a predictable way to interact with dropdowns and "other" options.  The pop up model (like for adding a Plant Community) is our standard.  Is it possible to make the indicator plant lists and actions function in the same way? It would mean having the custom value appear in the same space that would otherwise house the selected option from the ref table (at least I think where that value is coming from). Is that doable?

**Comment by LisaMoore1** [comment]
@calebissharp 
Typing in the top of the dropdown box filters which is good (I like this better than the pasture name layout)
If I select "other" from the list (whether the full or filtered list) I'm not prompted to enter new text to define what "other" is
If I enter text that isn't in the list and then leave the field (tab or click) the selection changes to "other" rather than the text (ex. xyz) that I entered.
I do like that if I type anything that isn't in the list it shows me "other" -- I think it just needs to either save the non-list text I entered or prompt me for a name if I select other.

**Comment by LisaMoore1** [comment]
@calebissharp Actually I think what is happening is that if I hit enter it is saving/showing the text (and returning cursor to beginning of entry) but if I just tab or touch elsewhere it is reverting to whatever was there. 
Preferred would be:
- exit field by tab/click = save what was entered 
- press ENTER = text saved & cursor moves to end of text or advances to next field (preferably advances)

**Comment by calebissharp** [comment]
@LisaMoore1 I can remove the "Other" option since the ability to add a custom plant supersedes that. My thinking was that if people just want to select "other" without providing a name, but I see now looking back that doesn't make much sense :)

The way the component is designed to work is that you have to explicitly select the "**Other**: different plant type" option by clicking it or pressing enter. Similar to if it was just a dropdown that had search. If you typed in part of a query, you wouldn't expect it to select the first entry if you click away. To me, clicking away is a "cancel" action (like with a dialog). Perhaps calling it a dropdown/textfield combination was a bit misleading, since it's more like a dropdown that allows you to add options. In this case you can only add one option.

For example:
1. User starts typing the name of a plant into the dropdown. 
2. They see it isn't a predefined option because it appears in the list as "**Other**: name I typed".
3. They press enter or click the option to select it, which sets the the text of the dropdown to the value they typed in.
4. Focus remains on the component and they can press tab to go to the next field **OR** start typing again and repeat steps 1-4.

It **is** possible to implement "exit field by tab/click = save what was entered" though (I think). Let me know if that's still your preference.

**Comment by LisaMoore1** [comment]
@calebissharp The "Other: name I typed" thing was really clear. I just couldn't figure out what I had to do to make the thing I typed to "stick" since data entry speedsters typically type text in a field then hit TAB (or click in the next field) to enter in the next field. This not being a regular text field it behaves differently but it took me a bit to stumble into "ENTER" being what made the text stick -- and then still wasn't sure because the cursor reset to the front of the field so I wasn't confident until I exited the field altogether.  

I think what I'm saying is that if it isn't crazy effort consumptive doing the "exit field by tab/click = save" would be preferred. Basically if the entry is there it has to have a value so if you want something different you just change it. If you don't want the entry you would delete it.

**Comment by LisaMoore1** [comment]
@calebissharp This played with my mind again today. First I went looking for an "other" option -- then I remembered I could just start typing. When I typed I once again expected that TAB would enter my text and move me to the next field. Can we update this so that:

- [ ] ENTER updates the text in the field
- [ ] TAB updates the text in the field (tab behaves like ENTER)
- [ ] either TAB or ENTER moves the cursor to the next field after updating the entered text
